data_discovery_ai/pipeline.py

```diff
@@ -54,20 +66,34 @@ def __init__(
         self.params = params
         self.isDataChanged = isDataChanged
         self.usePretrainedModel = usePretrainedModel
-        self.model_name = model_name
-        self.model = None
-        # TODO: needs to define what are the accepted values
-        #  validate against the list declared in data_discovery_ai/common/constants.py? needs to be a controlled list of values
-        if self.usePretrainedModel and self.model_name is None:
-            raise ValueError("model name should be given to use pretrained model")
+        # validate model name with accepted values, defined in data_discovery_ai/common/constants.py
+        self.validate_model_name(model_name=model_name)
+
+    """
+        Validate model name within fixed selections
+        Input:
+            model_name: str. The file name of the saved model. restricted within four options: development, staging, production, and test
+    """
+
+    def validate_model_name(self, model_name) -> None:
```

Comment:
validate method should return boolean, true or false or raise Error, not doing extra works of assigning value etc.
so you might consider changing method name to is_valid_modal() -> bool